tests/data/utils.py

```diff
@@ -24,7 +24,14 @@ def iterate_over_rules():
         leaf_dir = os.path.basename(dir_name)
         rel_dir = '.' + dir_name[len(_DIR):]
         if leaf_dir.startswith('rule_'):
-            result = Rule(rel_dir, _SSG_PREFIX + leaf_dir, files)
+            # Use only .sh files
+            # We care only about '.sh' files because only these files
+            # are supposed to contain bash scripts with test cases
+            # and we can expect them to be executable.
+            # Other files in rule directories are editor swap files
+            # or other content than a test case.
```

Comment:
I don't want to be nitpicky but if it was me I would change the "files" description in the doctext to say that it's a list of executable test-cases, maybe even rename it from files to testcases.

The comment itself is way too verbose, it could just say "Filter out everything except the shell test-cases". All the extra info you have in the comment should be in the doctext.